C compiler error when using length of var openArray argument as a default value for an argument

Nim fails to generate valid C code if one uses the length of an var openArray argument as a default value of another argument.

### Example
<!-- Paste your example in the code-block below. -->
```nim
func test*[T](input: var openArray[T], start: int = 0, fin: int = input.len - 1) =
    discard

var someSeq = @[1'i32]

test(someSeq)
```


### Current Output
```
CC: crash
Error: execution of an external compiler program 'gcc -c  -w  -I/home/clyybber/builds/nim/lib -I/home/clyybber/projects/misc/smoothsort -o /home/clyybber/.cache/nim/crash_d/crash.c.o /home/clyybber/.cache/nim/crash_d/crash.c' failed with exit code: 1

/home/clyybber/.cache/nim/crash_d/crash.c: In function ‘NimMainModule’:
/home/clyybber/.cache/nim/crash_d/crash.c:205:41: error: ‘colontmp_Len_0’ undeclared (first use in this function); did you mean ‘colontmp_’?
  TM_hjnR9bn9cEWJywRfQ2E5uL5A_5 = subInt(colontmp_Len_0, ((NI) 1));
                                         ^~~~~~~~~~~~~~
                                         colontmp_
/home/clyybber/.cache/nim/crash_d/crash.c:205:41: note: each undeclared identifier is reported only once for each function it appears in
```


### Expected Output
<!-- What should be the correct output? -->
Should compile


### Additional Information
<!--- For Example:
* Your Nim version (output of `nim -v`).
* Was it working in the previous Nim releases?
* A link to a related issue or discussion.
-->
```
$ nim -v
Nim Compiler Version 0.19.9
```

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -1882,8 +1882,17 @@ proc genArrayLen(p: BProc, e: PNode, d: var TLoc, op: TMagic) =
         if op == mHigh: unaryExpr(p, e, d, "($1Len_0-1)")
         else: unaryExpr(p, e, d, "$1Len_0")
       else:
-        if op == mHigh: unaryExpr(p, e, d, "($1.Field1-1)")
-        else: unaryExpr(p, e, d, "$1.Field1")
+        let isDeref = a.kind in {nkHiddenDeref, nkDerefExpr}
+        if op == mHigh:
+          if isDeref:
+            unaryExpr(p, e, d, "($1->Field1-1)")
+          else:
+            unaryExpr(p, e, d, "($1.Field1-1)")
+        else:
+          if isDeref:
+            unaryExpr(p, e, d, "$1->Field1")
+          else:
+            unaryExpr(p, e, d, "$1.Field1")
   of tyCstring:
     if op == mHigh: unaryExpr(p, e, d, "($1 ? (#nimCStrLen($1)-1) : -1)")
     else: unaryExpr(p, e, d, "($1 ? #nimCStrLen($1) : 0)")

--- a/tests/ccgbugs/t10964.nim
+++ b/tests/ccgbugs/t10964.nim
@@ -1,0 +1,6 @@
+func test*(input: var openArray[int32], start: int = 0, fin: int = input.len - 1) =
+    discard
+
+var someSeq = @[1'i32]
+
+test(someSeq)

